docs/pluggable-discovery-specification.md

```diff
@@ -377,3 +377,67 @@ myboard.upload_port.1.apples=40
 
 will match on both `pears=20, apples=30` and `pears=30, apples=40` but not `pears=20, apples=40`, in that sense each
 "set" of identification properties is independent from each other and cannot be mixed for port matching.
+
+#### Identification of board options
+
+The board identification can also identify [custom board options](platform-specification.md#custom-board-options) if the
+corresponding `upload_port` properties are defined:
```

Comment:
```suggestion
[Custom board options](platform-specification.md#custom-board-options) can also be identified.

Identification property values are associated with a custom board option by the board definition in
[`boards.txt`](platform-specification.md#boardstxt). Two formats are available.
```

- Adjust wording and formatting for readability
- Specify the association is done in `boards.txt` earlier in the documentation.
- Add reference link to `boards.txt` specification.